packages/spor-button-react-native/src/Button.tsx

```diff
@@ -7,69 +7,156 @@ import {
 };
 
 /** A button. */
 export const Button = ({
+  variant,
   onPress,
-  children,
-  isDisabled = false,
+  disabled = false,
   isLoading = false,
-  leftIcon,
-  rightIcon,
+  accessibilityLabel,
+  children,
   ...rest
-}: Props) => {
-  if (isDisabled) {
-    rest.variant = "disabled";
-  }
-  const { style, ...viewProps } = useRestyle(restyleFunctions, rest);
+}: ButtonProps) => {
+  const [pressed, setPressed] = useState(false);
 
-  const fontSize = findProperty("fontSize", style as any[]);
-  const fontWeight = findProperty("fontWeight", style as any[]);
-  const color = findProperty("color", style as any[]);
+  const { style: restyleStyle } = useRestyle(restyleFunctions, {
+    variant,
+    ...rest,
+  });
+  const activeStyle = pressed ? getActiveStyle(variant) : {};
+  const disabledAndLoadingStyle =
+    disabled || isLoading ? getDisabledAndLoadingStyle() : {};
+
+  const style = [
+    restyleStyle,
+    activeStyle,
+    disabledAndLoadingStyle,
+  ] as StyleProp<TextStyle>;
+
+  const flatStyles = StyleSheet.flatten(style);
+  const fontSize = flatStyles.fontSize;
+  const fontWeight = flatStyles.fontWeight;
+  const color = flatStyles.color;
+  const backgroundColor = flatStyles.backgroundColor;
 
   return (
-    <TouchableOpacity onPress={onPress} disabled={isDisabled}>
-      <View style={style as any} {...viewProps}>
-        {isLoading ? (
-          <ActivityIndicator color={color} />
-        ) : (
-          <Text style={{ color, fontSize, fontWeight }}>{children}</Text>
-        )}
-      </View>
-    </TouchableOpacity>
+    <Pressable
+      accessibilityRole="button"
+      accessibilityState={{ disabled }}
+      onPressIn={() => setPressed(true)}
+      onPressOut={() => setPressed(false)}
+      onPress={onPress}
+      disabled={disabled}
+      accessibilityLabel={accessibilityLabel}
+      style={style}
+      android_ripple={
+        ANDROID_SUPPORTS_RIPPLE
+          ? {
+              color: backgroundColor,
+            }
+          : undefined
+      }
+      {...rest}
+    >
+      <Text style={{ color, fontSize, fontWeight, opacity: isLoading ? 0 : 1 }}>
+        {children}
+      </Text>
+      {isLoading && (
+        <Box
+          style={{
+            position: "absolute",
+            justifyContent: "center",
+            alignItems: "center",
+          }}
+        >
+          <LightInlineLoader height={"75%"} />
+        </Box>
+      )}
+    </Pressable>
   );
 };
 
-// TODO: Make this more pretty
-const findProperty = (propertyName: string, styles: any[]) => {
-  return styles
-    .map((style) => style[propertyName])
-    .reduce((prev, next) => (next ? next : prev)) as any;
-};
+function getDisabledAndLoadingStyle(): StyleProp<TextStyle> {
+  return {
+    color: tokens.color.alias.dimGrey.value,
+    backgroundColor: tokens.color.alias.silver.value,
+    borderWidth: 0,
+  };
+}
+
+function getActiveStyle(variant: ButtonVariant): StyleProp<ViewStyle> {
```

Comment:
Det hadde vært litt digg å ha dette i theme-filen. Legg til en ny theme-key `buttonVariantsActive` f.eks. og bare legg til disse fargene der.

Så kan du f.eks. hente ut themet via `useTheme` hooken og applye stilene her istedenfor. Da slipper du en direkte avhengighet til tokens, og vi kan endre themet runtime down the line.